electron/shortcuts/nonsteamgame.ts

```diff
@@ -0,0 +1,371 @@
+      error,
+      adding: true
+    })
+  } else {
+    const errors = []
+    let added = false
+    for (const folder of folders) {
+      // skip this folders, because there are no steam user
+      if (folder === '0' || folder === 'ac') {
+        continue
+      }
+
+      const configDir = join(props.steamUserdataDir, folder, 'config')
+      const shortcutsFile = join(configDir, 'shortcuts.vdf')
+
+      if (!existsSync(configDir)) {
+        mkdirSync(configDir)
+      }
+
+      if (!existsSync(shortcutsFile)) {
+        writeShortcutFile(shortcutsFile, { shortcuts: [] })
+      }
+
+      // read file
+      const content = readShortcutFile(shortcutsFile)
+
+      const checkResult = checkIfShortcutObjectIsValid(content)
+      if (!checkResult.success) {
+        errors.push(
+          `Can't add "${props.gameInfo.title}" to steam user "${folder}". "${shortcutsFile}" is corrupted!`,
+          ...checkResult.errors
+        )
+        continue
+      }
+
+      if (checkIfAlreadyAdded(content, props.gameInfo.title) > -1) {
+        added = true
+        continue
+      }
+
+      // add new Entry
+      const newEntry = {} as ShortcutEntry
+      newEntry.AppName = props.gameInfo.title
+      newEntry.Exe = `"${app.getPath('exe')}"`
+
+      if (isFlatpak) {
+        newEntry.Exe = `"flatpak"`
+      } else if (process.env.APPIMAGE) {
+        newEntry.Exe = `"${process.env.APPIMAGE}"`
+      }
+
+      newEntry.StartDir = `"${process.cwd()}"`
+      await getIcon(props.gameInfo.app_name, props.gameInfo)
+        .then((path) => (newEntry.icon = path))
+        .catch((error) =>
+          logWarning(
+            `Couldn't find a icon for ${props.gameInfo.title} with: ${error}`
+          )
+        )
+
+      newEntry.LaunchOptions = `--no-gui --no-sandbox "heroic://launch/${props.gameInfo.app_name}"`
+      if (isFlatpak) {
+        newEntry.Exe = `run com.heroicgameslauncher.hgl ${newEntry.LaunchOptions}`
+      }
+      newEntry.IsHidden = false
+      newEntry.AllowDesktopConfig = true
+      newEntry.AllowOverlay = true
+      newEntry.OpenVR = false
+      newEntry.Devkit = false
+      newEntry.DevkitOverrideAppID = false
+
+      if (tsStore.has(`${props.gameInfo.app_name}.lastPlayed`)) {
+        newEntry.LastPlayTime = tsStore.get(
+          `${props.gameInfo.app_name}.lastPlayed`
+        ) as Date
+      } else {
+        newEntry.LastPlayTime = new Date()
+      }
+
+      content.shortcuts.push(newEntry)
+
+      // rewrite shortcuts.vdf
+      writeShortcutFile(shortcutsFile, content)
+
+      added = true
+    }
+
+    if (!added) {
+      showErrorInFrontend({
+        gameTitle: props.gameInfo.title,
+        error: errors.join('\n'),
+        adding: true
+      })
+    }
+
+    if (errors.length === 0) {
+      const message = `${props.gameInfo.title} was successfully added to steam.`
+      logInfo(message, LogPrefix.Shortcuts)
+      notifyFrontend({ message, adding: true })
```

Comment:
We should add the message to the translations since we are displaying it in the frontend.